lib/Basics/ReadWriteSpinLock.cpp

```diff
@@ -100,28 +96,14 @@ void ReadWriteSpinLock::lockWrite() noexcept {
 }
 
 bool ReadWriteSpinLock::lockWrite(std::size_t maxAttempts) noexcept {
-  if (tryLockWrite()) {
-    return true;
```

Comment:
This is a pessimization, now you need two modifications to get an uncontested lock

Reply:
Fixed